### DISCUSSION TITLE
router.refresh() does not bring new data to current route

### DISCUSSION THREAD

**Discussion by pedro757** [vercel/next.js#58856] [App Router] [answered]
I used revalidatePath() and then router.refresh() and I still see stale data in the current route

**Comment by ezeparziale** [comment]
Do you add the path inside the revalidatePath(`path`) ?

Example: revalidatePath("/users")

**Reply by pedro757** [reply]
Yes

**Reply by ezeparziale** [reply]
In dev mode try removing the .next folder

**Reply by pedro757** [reply, marked as answer]
Thank you! It was my fault u.u

I was using useState with a prop as defaultValue and the state wasn't actually changing until I made use of a useEffect to update the state when the prop changes.
